pkg/controller/worker/genericactuator/actuator.go

```diff
@@ -161,7 +161,7 @@ func (a *genericActuator) listMachineClassSecrets(ctx context.Context, namespace
 	var (
 		secretList = &corev1.SecretList{}
 		labels     = map[string]string{
-			v1beta1constants.GardenPurpose: GardenPurposeMachineClass,
+			v1beta1constants.GardenerPurpose: GardenPurposeMachineClass,
```

Comment:
This is a behavioural change, currently the machine class secrets are labelled with `garden.sapcloud.io/purpose`, this change tries to list them with `gardener.cloud/purpose`. Is it backwards compatible?
I would suggest to define here the `GardenPurpose` const to the deprecated label key and probably to handle the adaptation with other PR. WDYT?

Reply:
I've updated the `listMachineClassSecrets` implementation to handle this problem.